crates/conformance/src/spin_http.rs

```diff
@@ -21,9 +21,28 @@ pub(super) fn test(store: &mut Store<Context>, pre: &InstancePre<Context>) -> Re
             },
         )?;
 
-        ensure!(response.status == 200);
-        ensure!(response.headers == Some(vec![("foo".to_owned(), "bar".to_owned())]));
-        ensure!(response.body == Some(b"you said: Hello, SpinHttp!".to_vec()));
+        ensure!(
+            response.status == 200,
+            "expected response status 200, got {}",
+            response.status
+        );
+
+        ensure!(
+            response.headers == Some(vec![("foo".to_owned(), "bar".to_owned())]),
+            "expected a single response header, \"foo: bar\", got {:?}",
+            response.headers
+        );
+
+        let expected_body = "you said: Hello, SpinHttp!";
+
+        ensure!(
+            response.body == Some(expected_body.as_bytes().to_vec()),
+            "expected a response body containing the string \"{expected_body}\", got \"{:?}\"",
```

Comment:
> \"{expected_body}\"

For strings you get quotes (and escaping) for free from the debug flag, e.g. `{expected_body:?}`

> `\"{:?}\"`

This will double-double-quote the string.

Reply:
Good catch, thanks.